Apps/Src/DebugDump.c

```diff
@@ -57,6 +57,15 @@ static const char *FAULT_BITMAP_STRING[] = {
 };
 /*----------------------------------------------*/
 
+static void print_f(char * str, float f) {
+    printf(str);
+
+    int n = (int) f;
+    f -= n;
+    f *= (10 << 2);
+    printf("%d.%d\n\r", n, (int) f);
+}
```

Comment:
That function is broken, even if you change the multiply to be by 100 instead of 10 << 2 == 40. It doesn't handle negative numbers at all, for example.

Reply:
Updated to the version in motor_sim